release/train_tests/benchmark/train_benchmark.py

```diff
@@ -27,25 +31,34 @@ def __init__(self, factory: BenchmarkFactory):
         model = factory.get_model()
         self.model = ray.train.torch.prepare_model(model)
 
-        # TODO: Might want to re-initialize dataloaders at the start of every epoch.
-        self.train_dataloader = factory.get_train_dataloader()
-        self.val_dataloader = factory.get_val_dataloader()
-
         self.loss_fn = factory.get_loss_fn()
         self.optimizer = torch.optim.Adam(self.model.parameters(), lr=1e-3)
 
         # Training progress state.
         self._train_batch_idx: int = 0
         self._train_epoch_idx: int = 0
 
-        # Load checkpoint if available.
+        # Performance metrics
+        self._metrics = collections.defaultdict(lambda: Timer())
+
         checkpoint = ray.train.get_checkpoint()
         if checkpoint:
-            with checkpoint.as_directory() as temp_checkpoint_dir:
-                self.load_checkpoint(temp_checkpoint_dir)
+            self.restore_from_checkpoint(checkpoint)
 
-        # Performance metrics
-        self._metrics = collections.defaultdict(lambda: Timer())
+    def restore_from_checkpoint(self, checkpoint: ray.train.Checkpoint):
+        with tempfile.TemporaryDirectory(
+            dir="/mnt/local_storage"
+        ) as temp_checkpoint_dir:
+            download_start = time.perf_counter()
+            checkpoint.to_directory(temp_checkpoint_dir)
+            download_time = time.perf_counter() - download_start
+
+            load_start = time.perf_counter()
+            self.load_checkpoint(temp_checkpoint_dir)
+            load_time = time.perf_counter() - load_start
+
+            self._metrics["checkpoint/download"].add(download_time)
+            self._metrics["checkpoint/load"].add(load_time)
```

Comment:
nit: use same context manager pattern as done later with the batch iteration?

e.g.

```python
with self._metrics["checkpoint/download"].timer():
    checkpoint.to_directory(temp_checkpoint_dir)


with self._metrics["checkpoint/load"].timer():
    self.load_checkpoint(temp_checkpoint_dir)
```

Reply:
`load_checkpoint` will load in the snapshot of the metrics, which will overwrite these guys. So I need to save it separately then add it to the metrics later.